pkg/kubelet/prober/prober.go

```diff
@@ -206,11 +206,16 @@ func findPortByName(container api.Container, portName string) (int, error) {
 
 // formatURL formats a URL from args.  For testability.
 func formatURL(scheme string, host string, port int, path string) *url.URL {
-	return &url.URL{
-		Scheme: scheme,
-		Host:   net.JoinHostPort(host, strconv.Itoa(port)),
-		Path:   path,
+	u, err := url.Parse(path)
```

Comment:
Parse() also properly deals with the query string. `path` can/should be, in this case, a relative url without scheme or host (we override those anyway). I can add a comment explaining the reason for the Parse() call.